nodejs/week2/nodejsweekwork/src/backend/index.js

```diff
@@ -0,0 +1,41 @@
+const express = require("express");
+const router = express.Router();
+const app = express();
+
+const meals = require("./routes/meals");
+const reservations = require("./routes/reservations");
+const reviews = require("./routes/reviews.js");
+const cheapMeals = require("./routes/cheap-meals.js");
+const largeMeals = require("./routes/large-meals.js");
+const randommeal = require("./routes/randommeal.js");
+const randomReservation = require("./routes/randomReservation");
+
+const mealswithID = require("./routes/mealswithID.js");
+const reservationsWithID = require("./routes/reservationsWithID");
+const reviewsWithID = require("./routes/reviewsWithID.js");
+const mealPrice = require("./routes/mealPrice");
+const mealsTitle = require("./routes/mealsTitle");
+const mealsCreatedAfter = require("./routes/mealsCreatedAfter");
+const mealsLimit = require("./routes/mealsLimit");
+
+app.use("/meals", meals);
+app.use("/reservations", reservations);
+app.use("/reviews", reviews);
+app.use("/cheap-meals", cheapMeals);
+app.use("/large-meals", largeMeals);
+app.use("/randommeal", randommeal);
+app.use("/randomReservation", randomReservation);
+
+app.use("/meals", mealswithID);
+app.use("/reservationsid", reservationsWithID);
+app.use("/reviewsid", reviewsWithID);
+app.use("/meals/mealPrice", mealPrice);
```

Comment:
These routes are not correct. `mealPrice` , `title`, `createdAfter` and `limit` should be query params and not paths in a url.

The path that should handle all these query params is `/meals`. So one single request handler or router should be able to handle this request

`meals?mealPrice=100&title=hello&limit=4&createdAfter=2019-01-01`

you need to extract all those parameters and use them as filters within the same router/request handler.

Reply:
And its `maxPrice` instead of `mealPrice`.